test/tests/functional/pbs_holidays.py

```diff
@@ -71,15 +71,39 @@ def check_prime(self):
         j1 = Job(attrs=a)
         jid1 = self.server.submit(j1)
 
-        # Verify that the job does not run as it is 24hrs prime-time
+        # Verify that the job does not run as it is all prime-time
+        self.server.expect(JOB, {'comment': (MATCH_RE, '.*Not Running:.*')},
+                           jid1)
         self.server.expect(JOB, {'job_state': 'Q'}, jid1)
 
         # Submit a job to the normal default queue
         a = {ATTR_queue: self.p_queue}
         j2 = Job(attrs=a)
         jid2 = self.server.submit(j2)
 
-        # Verify that the job runs as it is 24hrs prime-time
+        # Verify that the job runs as it is all prime-time
+        self.server.expect(JOB, {'job_state': 'R'}, jid2)
+
+    def check_nonprime(self):
+        """
+        Helper function to test that it is non-prime time right now
```

Comment:
If the log_filter isn't filtering DEBUG2, the scheduler will tell you what prime status it is in and when it ends in the log.  Consider using the log message.